### PR TITLE
Tool for HRC CEA 

### PR DESCRIPTION
**Pull request by jzuhone**
## Description

This PR expands `acis_thermal_check` to be able to run HRC thermal models, in this case the CEA model. A breakdown:

* It adds the `cea_check` script necessary to customize the `ACISThermalCheck` class
* ~It adds custom HRC commanded states (H/T to @taldcroft)~ (this is now done in https://github.com/sot/kadi/pull/257)
* Validation plots for `2imonst`, `2sponst`, and `2s2onst`
* Small changes to plotting routines to support the above
* Removes dependence on CRM pad time file / CTI report file to determine rad exit and entry, uses the backstop commands now

Examples:

https://cxc.cfa.harvard.edu/acis/Thermal/tmp/cea_sep1922/ (includes recent HRC activity in prediction)
https://cxc.cfa.harvard.edu/acis/Thermal/tmp/cea_sep2622/ (includes recent HRC activity in validation)

I'd be interested in getting @drxre's take on the code in `_calc_model_supp` in particular.

## Interface impacts

A brand new interface for checking HRC temperatures!

## Testing

Unit testing has been set up

### Unit tests

- [ ] No unit tests
- [x] Mac
- [ ] Linux
- [ ] Windows

### Functional tests

Tested on SEP1922A and SEP2622A loads for correct prediction and validation plots.

### CODE REVIEW

**Comment by jzuhone** [comment]
ping @matthewdahmer @drxre

File: acis_thermal_check/apps/cea_check.py [hunk 1]
```diff
@@ -0,0 +1,142 @@
...
+        has an initial value (T_cea) - which it does at
+        prediction time (gets it from state0), then T_cea0 
+        is set to that.  If we are running the validation,
+        T_cea is set to None so we use the dvals in model.comp
+
+        NOTE: If you change the name of the dpa0 pseudo node you
+              have to edit the new name into the if statement
+              below.
+        """
+        if 'cea0' in model.comp:
```
[the remark this answers is not in the document]
Comment (jzuhone) [reply acis_thermal_check/apps/cea_check.py:87]: Both of these were oversights--fixing.

File: acis_thermal_check/apps/cea_check.py [hunk 2]
```diff
@@ -0,0 +1,142 @@
...
+        viols = {"hi":
+                     {"name": f"Hot ({self.limits['planning_hi'].value} C)",
+                      "type": "Max",
+                      "values": hi_viols}
+                 }
+        return viols
+
+    def _calc_model_supp(self, model, state_times, states, ephem, state0):
+        """
+        Update to initialize the cea0 pseudo-node. If 1dpamzt
```
[the remark this answers is not in the document]
Comment (jzuhone) [reply acis_thermal_check/apps/cea_check.py:77]: Thanks, fixing

**Comment by taldcroft** [comment]
@jzuhone - ah... OK, it worked with the `hrc` state builder. 👍

**Comment by taldcroft** [comment]
I'm making a quick PR to put those HRC states into kadi since they are of general use (e.g. quickly find when HRC was on). With that you should take out the custom states and these will need to be promoted together in the next Ska release (which will be relatively soon).

**Comment by jzuhone** [comment]
@taldcroft :+1:

**Comment by taldcroft** [comment]
@jzuhone - See https://github.com/sot/kadi/pull/257. You should be able to re-baseline this PR to use that and do "integration" testing of the two patches together. There is a functional test checkbox in 257 to click when it all works.

**Review by matthewdahmer — APPROVED** [review]
Looking at the `_calc_model_supp` function, the initialization looks good.

**Comment by taldcroft** [comment]
Is this going to be merged by Monday? We are freezing the content for Ska3-flight / matlab on Monday Nov 7.

**Comment by drxre** [comment]
I've successfully run this and know how to integrate it into my weekly review. I am OK with this.

**Pull request merged**